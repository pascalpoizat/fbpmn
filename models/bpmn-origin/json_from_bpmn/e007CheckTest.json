{
    "isInterrupt": {},
    "catN": {
        "Task_1eirt50": "AbstractTask",
        "Task_1rt44mz": "AbstractTask",
        "StartEvent_1": "NoneStartEvent",
        "EndEvent_0v9lt5i": "NoneEndEvent",
        "ExclusiveGateway_0079typ": "XorGateway",
        "Task_097548f": "AbstractTask",
        "ExclusiveGateway_1j1chqb": "AndGateway",
        "A_": "Process"
    },
    "catE": {
        "SequenceFlow_0z2xwql": "NormalSequenceFlow",
        "SequenceFlow_0uzla8o": "NormalSequenceFlow",
        "SequenceFlow_0wto9d1": "NormalSequenceFlow",
        "SequenceFlow_0b7efwa": "NormalSequenceFlow",
        "SequenceFlow_0uplc1a": "NormalSequenceFlow",
        "SequenceFlow_1fuwd1z": "NormalSequenceFlow",
        "SequenceFlow_01wc4ks": "NormalSequenceFlow"
    },
    "containN": {
        "A_": [
            "Task_097548f",
            "Task_1eirt50",
            "ExclusiveGateway_0079typ",
            "EndEvent_0v9lt5i",
            "Task_1rt44mz",
            "ExclusiveGateway_1j1chqb",
            "StartEvent_1"
        ]
    },
    "messageE": {},
    "timeInformation": {},
    "targetE": {
        "SequenceFlow_0z2xwql": "ExclusiveGateway_1j1chqb",
        "SequenceFlow_0uzla8o": "EndEvent_0v9lt5i",
        "SequenceFlow_0wto9d1": "Task_1eirt50",
        "SequenceFlow_0b7efwa": "ExclusiveGateway_0079typ",
        "SequenceFlow_0uplc1a": "Task_097548f",
        "SequenceFlow_1fuwd1z": "Task_1rt44mz",
        "SequenceFlow_01wc4ks": "ExclusiveGateway_0079typ"
    },
    "name": "e007CheckTest",
    "attached": {},
    "containE": {
        "A_": [
            "SequenceFlow_0z2xwql",
            "SequenceFlow_0uzla8o",
            "SequenceFlow_1fuwd1z",
            "SequenceFlow_0b7efwa",
            "SequenceFlow_0wto9d1",
            "SequenceFlow_01wc4ks",
            "SequenceFlow_0uplc1a"
        ]
    },
    "edges": [
        "SequenceFlow_0z2xwql",
        "SequenceFlow_0uzla8o",
        "SequenceFlow_1fuwd1z",
        "SequenceFlow_0b7efwa",
        "SequenceFlow_0wto9d1",
        "SequenceFlow_01wc4ks",
        "SequenceFlow_0uplc1a"
    ],
    "messages": [],
    "sourceE": {
        "SequenceFlow_0z2xwql": "StartEvent_1",
        "SequenceFlow_0uzla8o": "Task_1rt44mz",
        "SequenceFlow_0wto9d1": "ExclusiveGateway_1j1chqb",
        "SequenceFlow_0b7efwa": "Task_1eirt50",
        "SequenceFlow_0uplc1a": "ExclusiveGateway_1j1chqb",
        "SequenceFlow_1fuwd1z": "ExclusiveGateway_0079typ",
        "SequenceFlow_01wc4ks": "Task_097548f"
    },
    "nodes": [
        "A_",
        "Task_097548f",
        "Task_1eirt50",
        "ExclusiveGateway_0079typ",
        "EndEvent_0v9lt5i",
        "Task_1rt44mz",
        "ExclusiveGateway_1j1chqb",
        "StartEvent_1"
    ],
    "nameN": {
        "Task_1eirt50": "Task A",
        "Task_1rt44mz": "Task C",
        "Task_097548f": "Task B"
    }
}