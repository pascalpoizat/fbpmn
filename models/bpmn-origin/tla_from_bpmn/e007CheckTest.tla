---------------- MODULE e007CheckTest ----------------

EXTENDS TLC, PWSTypes

VARIABLES nodemarks, edgemarks, net

Interest ==
  "A_" :> {  }

ContainRel ==
  "A_" :> { "Task_097548f", "Task_1eirt50", "ExclusiveGateway_0079typ", "EndEvent_0v9lt5i", "Task_1rt44mz", "ExclusiveGateway_1j1chqb", "StartEvent_1" }

Node == { "A_", "Task_097548f", "Task_1eirt50", "ExclusiveGateway_0079typ", "EndEvent_0v9lt5i", "Task_1rt44mz", "ExclusiveGateway_1j1chqb", "StartEvent_1" }

Edge == { "SequenceFlow_0z2xwql", "SequenceFlow_0uzla8o", "SequenceFlow_1fuwd1z", "SequenceFlow_0b7efwa", "SequenceFlow_0wto9d1", "SequenceFlow_01wc4ks", "SequenceFlow_0uplc1a" }

Message == {  }

msgtype ==
  [ i \in {} |-> {}]


source ==
   "SequenceFlow_0z2xwql" :> "StartEvent_1"
@@ "SequenceFlow_0uzla8o" :> "Task_1rt44mz"
@@ "SequenceFlow_1fuwd1z" :> "ExclusiveGateway_0079typ"
@@ "SequenceFlow_0b7efwa" :> "Task_1eirt50"
@@ "SequenceFlow_0wto9d1" :> "ExclusiveGateway_1j1chqb"
@@ "SequenceFlow_01wc4ks" :> "Task_097548f"
@@ "SequenceFlow_0uplc1a" :> "ExclusiveGateway_1j1chqb"

target ==
   "SequenceFlow_0z2xwql" :> "ExclusiveGateway_1j1chqb"
@@ "SequenceFlow_0uzla8o" :> "EndEvent_0v9lt5i"
@@ "SequenceFlow_1fuwd1z" :> "Task_1rt44mz"
@@ "SequenceFlow_0b7efwa" :> "ExclusiveGateway_0079typ"
@@ "SequenceFlow_0wto9d1" :> "Task_1eirt50"
@@ "SequenceFlow_01wc4ks" :> "ExclusiveGateway_0079typ"
@@ "SequenceFlow_0uplc1a" :> "Task_097548f"

CatN ==
   "A_" :> Process
@@ "Task_097548f" :> AbstractTask
@@ "Task_1eirt50" :> AbstractTask
@@ "ExclusiveGateway_0079typ" :> ExclusiveOr
@@ "EndEvent_0v9lt5i" :> NoneEndEvent
@@ "Task_1rt44mz" :> AbstractTask
@@ "ExclusiveGateway_1j1chqb" :> Parallel
@@ "StartEvent_1" :> NoneStartEvent

CatE ==
   "SequenceFlow_0z2xwql" :> NormalSeqFlow
@@ "SequenceFlow_0uzla8o" :> NormalSeqFlow
@@ "SequenceFlow_1fuwd1z" :> NormalSeqFlow
@@ "SequenceFlow_0b7efwa" :> NormalSeqFlow
@@ "SequenceFlow_0wto9d1" :> NormalSeqFlow
@@ "SequenceFlow_01wc4ks" :> NormalSeqFlow
@@ "SequenceFlow_0uplc1a" :> NormalSeqFlow

LOCAL preEdges ==
  [ i \in {} |-> {}]
PreEdges(n,e) == preEdges[n,e]

PreNodes(n,e) == { target[ee] : ee \in preEdges[n,e] }
          \union { nn \in { source[ee] : ee \in preEdges[n,e] } : CatN[nn] \in { NoneStartEvent, MessageStartEvent } }

BoundaryEvent ==
  [ i \in {} |-> {}]

WF == INSTANCE PWSWellFormed
ASSUME WF!WellTyped
ASSUME WF!WellFormedness

ConstraintNode == TRUE \* none
ConstraintEdge == TRUE \* none
Constraint == TRUE     \* none
INSTANCE PWSConstraints
INSTANCE UserProperties
INSTANCE PWSSemantics

================================================================

