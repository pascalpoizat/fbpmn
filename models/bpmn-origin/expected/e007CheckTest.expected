Working in /tmp/e007CheckTest.NwpfY with 1 worker(s)
[92mtransformation done[0m
<<"Processes=", 1, "Nodes=", 8, "Gateway=", 2, "SF=", 7, "MF=", 0>>
---------- Network01Bag ----------
[X] Prop01Type
     states=44 trans=73 depth=15
[ ] Prop02Safe
     states= trans= depth=9
[ ] Prop03Sound
     states=44 trans=73 depth=
[ ] Prop04MsgSound
     states=44 trans=73 depth=
---------- Network02FifoPair ----------
[X] Prop01Type
     states=44 trans=73 depth=15
[ ] Prop02Safe
     states= trans= depth=9
[ ] Prop03Sound
     states=44 trans=73 depth=
[ ] Prop04MsgSound
     states=44 trans=73 depth=
---------- Network03Causal ----------
[X] Prop01Type
     states=44 trans=73 depth=15
[ ] Prop02Safe
     states= trans= depth=9
[ ] Prop03Sound
     states=44 trans=73 depth=
[ ] Prop04MsgSound
     states=44 trans=73 depth=
---------- Network04Inbox ----------
[X] Prop01Type
     states=44 trans=73 depth=15
[ ] Prop02Safe
     states= trans= depth=9
[ ] Prop03Sound
     states=44 trans=73 depth=
[ ] Prop04MsgSound
     states=44 trans=73 depth=
---------- Network05Outbox ----------
[X] Prop01Type
     states=44 trans=73 depth=15
[ ] Prop02Safe
     states= trans= depth=9
[ ] Prop03Sound
     states=44 trans=73 depth=
[ ] Prop04MsgSound
     states=44 trans=73 depth=
---------- Network06Fifo ----------
[X] Prop01Type
     states=44 trans=73 depth=15
[ ] Prop02Safe
     states= trans= depth=9
[ ] Prop03Sound
     states=44 trans=73 depth=
[ ] Prop04MsgSound
     states=44 trans=73 depth=
---------- Network07RSC ----------
[X] Prop01Type
     states=44 trans=73 depth=15
[ ] Prop02Safe
     states= trans= depth=9
[ ] Prop03Sound
     states=44 trans=73 depth=
[ ] Prop04MsgSound
     states=44 trans=73 depth=
done.
